mem0/vector_stores/azure_ai_search.py

```diff
@@ -126,39 +161,41 @@ def insert(self, vectors, payloads=None, ids=None):
     def _build_filter_expression(self, filters):
         filter_conditions = []
         for key, value in filters.items():
-            # If the value is a string, add quotes
             if isinstance(value, str):
                 condition = f"{key} eq '{value}'"
             else:
                 condition = f"{key} eq {value}"
             filter_conditions.append(condition)
-        # Use 'and' to join multiple conditions
-        filter_expression = ' and '.join(filter_conditions)
+        filter_expression = " and ".join(filter_conditions)
```

Comment:
This only works for the payload properties that you pulled into fields in the index (user_id, run_id, agent_id), are custom properties expected in the payload? If so, those will need different handling.